### PR TITLE
Add prince 10.r7

### PR DESCRIPTION
**Pull request by Fryguy**
Prince PDF converter.

This was recommended to be moved to homebrew/binary in the discussion [here](https://github.com/Homebrew/homebrew/pull/25907), which the original author hasn't done in 2 weeks.  In there was also a question about the license which is [here](http://www.princexml.com/license/), though I'm not sure if I'm supposed to do anything about that.

### CODE REVIEW

**Comment by chrisopedia** [comment]
Any idea on if this will get merged anytime soon?

**Comment by deepak** [comment]
can make couple of updates

1. use the latest version ie. `10r2`
2. use sha256

so formula becomes

```ruby
require 'formula'

class Prince < Formula
  homepage 'http://www.princexml.com'
  url 'http://www.princexml.com/download/prince-10r2-macosx.tar.gz'
  sha256 'fcecdbd13e3758ecb96b7e8fb8c3ba0acef3e0c6c642e48ad084f1c6a3bd4fef'

  def install
    system "echo #{prefix} | ./install.sh"
  end
end
```

have tested it locally and it works

not sure why the build is failing. cannot login to the CI server

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

File: prince.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'formula'
```
Comment (bfontaine) [inline prince.rb]: You can remove this line.

File: prince.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+class Prince < Formula
+  homepage 'http://www.princexml.com'
```
Comment (bfontaine) [inline prince.rb]: Please use double quotes everywhere.

File: prince.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  url 'http://www.princexml.com/download/prince-9.0r4-macosx.tar.gz'
+  sha1 '692c1aff7081c8569ba6afb8341aba0788cbf456'
```
Comment (bfontaine) [inline prince.rb]: Please use a `sha256` here instead.

File: prince.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+  def install
+    system "echo #{prefix} | ./install.sh"
```
Comment (bfontaine) [inline prince.rb]: Would `shell_output("./install.sh", prefix.to_s)` work here?

**Comment by Fryguy** [comment]
Whoa...I didn't realize I had responses here...I'll see what I can do

**Comment by Fryguy** [comment]
@bfontaine Updated

The requested change to install to use shell_output doesn't seem to work as that method is not available in that context.

I am also getting some `brew audit` issue about openssl, but I don't know why, and adding `depends_on "openssl"` does nothing.  Is this important?

**Comment by bfontaine** [comment]
You can’t do anything about the `openssl` issue because this is a binary release; we’re not compiling it ourselves and thus can’t link to `openssl`.

**Comment by Fryguy** [comment]
I have failures here, but I can't see them (did it just get too old)?  How can I rekick them?

**Comment by Fryguy** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
@Fryguy It doesn’t work if you’re not a maintainer :wink:

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by Fryguy** [comment]
> @Fryguy It doesn’t work if you’re not a maintainer :wink:

haha..oh well I tried ;)

Ok, so I fixed the issue in question which was that the description was too long.  I *think* everything should be good to go now.  This also updates it to 10r7 (and I've updated the PR title).

**Comment by Fryguy** [comment]
The failure is

```text
==> audit problems
homebrew/binary/prince:
 * object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/prince/10r7/lib/prince/bin/prince
```

So, I'm not sure what to do about that :confused:

**Comment by bfontaine** [comment]
@Fryguy Don’t worry about this one; you can’t do anything because it’s already compiled.

**Comment by Fryguy** [comment]
Ok, then I think this one is good to go.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew, @Fryguy; we appreciate it :wink:

**Pull request closed without merging**